### ISSUE TITLE
Tabs API Change

### ISSUE THREAD

**Issue by acornejo** [react-toolbox/react-toolbox#739]
I was looking into the tab component just now, and I would like to propose a refactoring.

TL;DR the children elements of the tab should not represent the content that will be displayed when the tab is active.

This affords the user a lot more flexibility on how and where he chooses to place the actual content, for instance, you may want to have it above the tabbar, or below the tabbar. It would also solve a couple of outstanding issues, since if the content of the tabs has react components with state, then you may not want to remove them from the dom when inactive (the user can choose to instead hide the inactive content with css for instance).

I think making the content of the tabs part of the tabbar component is a bit like making the panel a child of the appbar instead of a sibling.

**Comment by javivelasco** [comment]
I agree with you. Tabs definitively need to be refactored and the API should change in the direction you've mentioned. In the first approach there were a lot of missing use cases like this. Tabs should be able to be displayed on the top of your app and providing a way to communicate with the external components to modify some other dependent component.

We can discuss in this issue how should be the perfect API. I'm leaving my proposal during the next days!

**Comment by acornejo** [comment]
Any thoughts here? With the new and shiny postcss release coming, I think its a good moment to make this API breaking change (also its easier for me to work on this during weekends, and its friday ;) ). 

PS: I also want to implement bottom navigator and bottom sheets (for the latter I wanted to refactor the layout component to allow it to come-up from the left, right, top and bottom).

**Comment by javivelasco** [comment]
Yeah, I think this can be almost re-written and the best would be to do it in the postcss branch. If you want to make some work and PR it to the postcss-branch it would be great! The Layout could just treat in a special way Drawer and AppBar so we can get rid of that duplication

**Comment by javivelasco** [comment]
@acornejo I've been thinking about this. Actually there is no need to refactor the API. The `label` property can receive a node to fully customize the tab content. If you don't want to specify the content for each tab you can just pass no children and the selected tab can be handled outside the component. This allows to decide what to place and where should be placed.

What would be the API with this refactor?
